Security.java

```diff
@@ -0,0 +1,101 @@
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.util.Base64;
+
+public class Security {
+	
+	private static Security single_instance = null;
+	
+	private Security() {		
+	}	
+	
+	public static Security getInstance() {
+		// thread safe, synchronized block
```

Comment:
Un dettaglio che non era richiesto ovviamente dall'esercizio, però non credo sia 'thread safe' questo metodo, in quanto il blocco non mi sembra syncronizzato, quindi è errato questo commento